fe/fe-core/src/main/java/org/apache/doris/job/extensions/mtmv/MTMVTask.java

```diff
@@ -230,7 +242,38 @@ public void run() throws JobException {
             MetaLockUtils.readLockTables(tableIfs);
             try {
                 context = MTMVRefreshContext.buildContext(mtmv);
-                this.needRefreshPartitions = calculateNeedRefreshPartitions(context);
+                if (mtmv.getIncrementalRefresh()) {
+                    Set<BaseTableInfo> baseTables = relation.getBaseTablesOneLevel();
+                    if (baseTables.size() != 1) {
+                        throw new RuntimeException("Only support incremental refresh for single table MTMV");
+                    }
+                    BaseTableInfo baseTableInfo = baseTables.iterator().next();
+                    MTMVSnapshotIf mvSnapshot = mtmv.getRefreshSnapshot().getMVSnapshot(mtmv.getName(), baseTableInfo);
+                    mvSnapshotId = Optional.ofNullable(mvSnapshot).map(MTMVSnapshotIf::getSnapshotVersion).orElse(0L);
+                    params.put(TableScanParams.READ_MODE, TableScanParams.INCREMENTAL_READ);
+                    if (taskContext.getTriggerMode() == MTMVTaskTriggerMode.MANUAL
+                            && (taskContext.isComplete() || !CollectionUtils.isEmpty(taskContext.getPartitions()))) {
+                        params.put(TableScanParams.DORIS_START_SNAPSHOT_ID, String.valueOf(mvSnapshotId));
```

Comment:
Try to decouple the logic of the previous partition refresh and the current incremental refresh as much as possible (you might consider splitting them into different classes instead of using a lot of if and else statements everywhere), and then add some comments.